### PR TITLE
Improve docs around redis and DISCOVERY_HOSTNAME

### LINKED ISSUE #160: Document the use of DISCOVERY_HOSTNAME environment variable
Certain (most?) redis implementations require setting the hostname explicitly in order for successful connections. We normally achieve this by setting `DISCOVERY_HOSTNAME`, and we should probably add it to the README.

### PR DESCRIPTION
**Pull request by drubin**
Also, improve docs around docker-cloud to highlight it's not supported

Closes #160

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1254,22 +1253,26 @@ Docker Swarm, in any cloud environment.
 ## Using centralized discovery tools
 
 cote is built to be zero-configuration, and relies on IP broadcast/multicast
-to work. However, as of 2017, cloud providers don't support this functionality
-out of the box. In these cases, one can use Docker Cloud and its Weave network
-integration. However, Docker Cloud may not be suitable for everyone, due to
+to work. cloud providers don't support this functionality (and they wont)
```
Comment (dashersw) [inline README.md]: `won't`

File: README.md [hunk 2]
```diff
@@ -1254,22 +1253,26 @@ Docker Swarm, in any cloud environment.
...
+out of the box. In these cases, one can use the Weave network overly
```
Comment (dashersw) [inline README.md]: `overlay`

File: README.md [hunk 3]
```diff
@@ -1254,22 +1253,26 @@ Docker Swarm, in any cloud environment.
...
-accomodate different solutions that can serve as the automated service
+accommodate different solutions that can serve as the automated service
 discovery tool. Currently, redis is supported out of the box, and cote
 makes use of the [node_redis](https://github.com/NodeRedis/node_redis)
 library, in case you want to use redis as the central discovery tool. If you
 need to use anything other than redis, please open
 [a new issue](https://github.com/dashersw/cote/issues/new) and we may be
 able to help.
 
+You will be should also set `DISCOVERY_HOSTNAME` to the **IP address**
```
Comment (dashersw) [inline README.md]: ~~will be should~~ should

File: README.md [hunk 4]
```diff
@@ -1254,22 +1253,26 @@ Docker Swarm, in any cloud environment.
...
+of the container/instance since it defaults to machines hostname which in
```
Comment (dashersw) [inline README.md]: `machine's`

File: README.md [hunk 5]
```diff
@@ -1290,10 +1293,11 @@ starts with `COTE_DISCOVERY_REDIS` to proper configuration for the
...
-| COTE_DISCOVERY_REDIS_URL    | Sets the redis connection URL. Has to start with either `redis://` or `//`. Enables the redis plugin.
-| COTE_DISCOVERY_REDIS_HOST   | Sets the redis connection host name. Enables the redis plugin.
-| COTE_DISCOVERY_REDIS_PORT   | Sets the redis connection port. Enables the redis plugin.
+| `COTE_DISCOVERY_REDIS`        | If you are running redis on localhost, setting this variable to true will use the locally available redis at port 6379. If you need any other redis URL or host, you don't need to use this variable.
+| `COTE_DISCOVERY_REDIS_URL`    | Sets the redis connection URL. Has to start with either `redis://` or `//`. Enables the redis plugin.
+| `COTE_DISCOVERY_REDIS_HOST`   | Sets the redis connection host name. Enables the redis plugin.
+| `COTE_DISCOVERY_REDIS_PORT`   | Sets the redis connection port. Enables the redis plugin.
+| `DISCOVERY_HOSTNAME`   | This defaults to your machines `hostname`. If this is not routable you need to set this to the routable IP address of this instance.
```
Comment (dashersw) [inline README.md]: `machine's`

File: README.md [hunk 6]
```diff
@@ -1302,7 +1306,7 @@ cote also supports other connection options supported by
 
 As an environment variable:
 ```sh
-COTE_DISCOVERY_REDIS_HOST=redis node service.js
+COTE_DISCOVERY_REDIS_HOST=redis DISCOVERY_HOSTNAME=127.0.0.1  node service.js
```
Comment (dashersw) [inline README.md]: single space after the IP

**Comment by drubin** [comment]
@dashersw updated thanks for reviewing.

**Pull request merged**